File: examples/webhooks/verify-sdk/verify-sdk.py
from flask import Flask
from flask import request
from bloock.client.webhook import WebhookClient

app = Flask(__name__)

SECRET_KEY = "NHJTAE6ikKBccSaeCSBSWGdp7NmixXy7"


@app.route('/verify', methods=['POST'])
def index():
    enforce_tolerance = False  # decide if you want to set tolerance when verifying

    body = request.data
    bloock_signature = request.headers['Bloock-Signature']

    webhook_client = WebhookClient()
    ok = webhook_client.verify_webhook_signature(
        body, bloock_signature, SECRET_KEY, enforce_tolerance)

    if not ok:
        raise ValueError('Invalid Signature!')
    else:
        print('Valid Signature!')

    return 'Finish'


app.run(host='0.0.0.0', port=81)

File: examples/webhooks/verify-sdk/verify-sdk.go
package main

import (
	"io"
	"log"
	"net/http"

	"github.com/bloock/bloock-sdk-go/v2/client"
)

// SecretKey represents the client secret key associate to your webhook endpoint
const SecretKey = "NHJTAE6ikKBccSaeCSBSWGdp7NmixXy7"

func main() {
	verifyHandler := func(w http.ResponseWriter, req *http.Request) {
		enforceTolerance := false // decide if you want to set tolerance when verifying
		body, err := io.ReadAll(req.Body)
		if err != nil {
			log.Fatalf("Cannot read body request: %v", err)
		}
		bloockSignature := req.Header.Get("Bloock-Signature")

		webhookClient := client.NewWebhookClient()
		ok, err := webhookClient.VerifyWebhookSignature(body, bloockSignature, SecretKey, enforceTolerance)
		if err != nil {
			log.Fatal(err)
		}
		if !ok {
			log.Fatal("Invalid Signature!")
		}
		log.Println("Valid Signature!")
	}

	http.HandleFunc("/verify", verifyHandler)
	log.Println("Listing for requests at http://localhost:8000/verify")
	log.Fatal(http.ListenAndServe(":8000", nil))
}
